core_backend/app/llm_call/check_output.py

```diff
@@ -41,71 +46,85 @@ def check_align_score(func: Callable) -> Callable:
+                )
+            )
+            return llm_response
+        else:
+            return await _check_align_score(llm_response)
 
     return wrapper
 
 
-async def _check_align_score(llm_response: UserQueryResponse) -> UserQueryResponse:
+async def _check_align_score(
+    llm_response: UserQueryResponse,
+) -> UserQueryResponse | UserQueryResponseError:
     """
     Check the alignment score
     """
+
     evidence = _build_evidence(llm_response)
     claim = llm_response.llm_response
-    if claim is not None:
-        payload = Payload(evidence=evidence, claim=claim)
+    assert claim is not None, "LLM response is None"
+    align_score_date = AlignScoreData(evidence=evidence, claim=claim)
 
-        if ALIGN_SCORE_METHOD is None:
-            logger.warning(
-                "No alignment score method specified but `check_align_score` was called"
-            )
-            return llm_response
-
-        elif ALIGN_SCORE_METHOD == "AlignScore":
-            align_score = await _get_alignScore_score(ALIGN_SCORE_API, payload)
-        elif ALIGN_SCORE_METHOD == "LLM":
-            align_score = await _get_llm_align_score(payload)
-        else:
-            raise NotImplementedError(f"Unknown method {ALIGN_SCORE_METHOD}")
-
-        if align_score.score < float(ALIGN_SCORE_THRESHOLD):
-            llm_response.llm_response = STANDARD_FAILURE_MESSAGE
-            llm_response.state = ResultState.ERROR
+    if ALIGN_SCORE_METHOD is None:
+        logger.warning(
+            "No alignment score method specified but `check_align_score` was called"
+        )
+        return llm_response
 
-        llm_response.debug_info["factual_consistency"] = {
-            "method": ALIGN_SCORE_METHOD,
-            "score": align_score.score,
-            "reason": align_score.reason,
-        }
+    elif ALIGN_SCORE_METHOD == "AlignScore":
+        align_score = await _get_alignScore_score(ALIGN_SCORE_API, align_score_date)
+    elif ALIGN_SCORE_METHOD == "LLM":
+        align_score = await _get_llm_align_score(align_score_date)
     else:
-        logger.warning(
-            (
-                "No LLM response found in the LLM response but "
-                "`check_align_score` was called"
-            )
+        raise NotImplementedError(f"Unknown method {ALIGN_SCORE_METHOD}")
+
+    factual_consistency = {
+        "method": ALIGN_SCORE_METHOD,
+        "score": align_score.score,
+        "reason": align_score.reason,
+    }
+
+    if align_score.score < float(ALIGN_SCORE_THRESHOLD):
+        error_response = UserQueryResponseError(
+            query_id=llm_response.query_id,
+            error_message=STANDARD_FAILURE_MESSAGE,
+            error_type=ErrorType.ALIGNMENT_TOO_LOW,
+            debug_info=llm_response.debug_info.copy(),
         )
-    return llm_response
+        error_response.debug_info["factual_consistency"] = factual_consistency.copy()
+        return error_response
```

Comment:
I did some more testing and this means that if the LLM response is not good, the user doesn't even get the option of reading the source FAQs (because we responded with an Error 400 without any of the usual payload). I'm not sure if this is what we want. We should probably either:

1. Go with this as it is but remove the default `STANDARD_FAILURE_MESSAGE` for the LLM and return Null (since that always triggers this guardrail anyway), or
2. Remove this error catching step and return a 200 + `STANDARD_FAILURE_MESSAGE` + FAQs so botpress can show the FAQs to the user anyway.